keyboards/anne_pro/anne_pro_lighting.c

```diff
@@ -192,3 +196,35 @@ void anne_pro_lighting_set_keys(uint8_t keys, uint8_t *payload) {
     uart_tx_ringbuf_write(&led_uart_ringbuf, 5, buf);
     uart_tx_ringbuf_write(&led_uart_ringbuf, 5 * keys, payload);
 }
+
+/* Set Caps Lock red led on */
+void anne_pro_lighting_caps_lock_on(){
+		if(!leds_enabled && !not_init){
+			/* Wake up the LED controller if off */
+			writePinHigh(C15);
+			chThdSleepMilliseconds(50);
+			not_init=true;
+		}
+		if(leds_enabled) // solves problems when LT(1,KC_CAPS) is used in keymap(Tap = caps, Hold = MO(1))
+			anne_pro_lighting_update();
+		uart_tx_ringbuf_write(&led_uart_ringbuf,4,"\x09\x0c\x0c\x01");
```

Comment:
I'm not sure how you came up with this packet, but it seems incorrect. First of all the packet length is marked as 12 (second byte), while the packet length is clearly only 2. Also the combination of 9 (LED) and 12 (GetThemeId) seems wrong. Please let me know how you came to this packet!

Reply:
I know its wrong, i got to it by chance when trying to do it following the reverse engineered info.Still trying to find the correct way to do it

Reply:
Also where do u get that 12 is GetThemeId, [](https://github.com/leonardlorenz/anne-pro-community-app/blob/master/com/obins/anne/utils/Msgprotocol.java) , says its 8. if there is any other site i ca get info , much appreciated.

Reply:
changed to uart_tx_ringbuf_write(&led_uart_ringbuf,4,"\x09\x02\x0c\x01");